modules/bim/db/migrate/20211011204301_add_column_conversion_status_to_ifc_model.rb

```diff
@@ -0,0 +1,28 @@
+class AddColumnConversionStatusToIfcModel < ActiveRecord::Migration[6.1]
+  def up
+    add_column(:ifc_models, :conversion_status, :integer, default: 0) # default "pending"
+    add_column(:ifc_models, :conversion_error_message, :text)
+
+    # Set IfcModels having a XKT file as "processed"
+    ::Bim::IfcModels::IfcModel
+      .joins(:attachments)
+      .where("attachments.description = 'xkt'")
+      .update_all(conversion_status: 2) # 2 == processed
+
+    # Set IfcModels not having a XKT file as conversion_status "error"
+    processed_model_ids = ::Bim::IfcModels::IfcModel
+                            .joins(:attachments)
+                            .where("attachments.description = 'xkt'")
+                            .pluck(:id)
+
+    ::Bim::IfcModels::IfcModel
+      .where
+      .not("id IN (?)", processed_model_ids)
+      .update_all(conversion_status: 3) # 3 == error
```

Comment:
I initially thought it faulty. It might become easier to read if it where changed like this:

```
processed_models = ::Bim::IfcModels::IfcModel
                   .joins(:attachments)
                   .where(attachments: { description: 'xkt' })
unprocessed_models = ::Bim::IfcModels::IfcModel
                     .where
                     .not(id: processed_models)

processed_models.update_all(conversion_status: 2)
unprocessed_models.update_all(conversion_status: 3)
```

Reply:
I improved that.